### PR TITLE
[TS] LPS-154862 Unable to manage a site without the user locale enabled

### PR DESCRIPTION
**Pull request by joseluisbango**
Relevant issues: 

- https://issues.liferay.com/browse/LPS-154862
- https://issues.liferay.com/browse/PTR-3085

---

Hi @liferay-core-infra,

Could you review this pull?

As we discussed in PTR-3085 with Lima and Echo teams, users should be able to manage a Site using any language as long as it is available in the instance. Currently, they can only do that if they go to Site Admin from _Control Panel > Sites_. If they navigate to Site Admin from a different place, such us the Site Selector shown when clicking the compass icon (what they usually do), they cannot change the display language.

They were able to do this in 6.2 manually adding the language code to the URL, so they perceive that as loss of functionality.

I have added a property as requested in the PTR so that the users can see the Site Admin pages (`/~/control_panel` pages) in any instance-available language. In addition, I have changed the `user_locale_options.jsp` to allow the users to easily change to their language.

Let me know if you have any questions.

Thanks!

### CODE REVIEW

**Comment by dantewang** [comment]
@joseluisbango I'll look into this one tomorrow, but I'm a bit confused by the message of this commit: https://github.com/liferay-core-infra/liferay-portal/pull/961/commits/4ce728c0bd0c97ac9a3fabecac7a40ad7203f3f4.

Does master need it, or is this only for 7.2.x? For the former, please modify the message; for the latter, please exclude the commit.

**Comment by joseluisbango** [comment]
Hey @dantewang,
I added that commit because I followed the approach used in the latest modification of that file: 
- This is the commit: https://github.com/liferay/liferay-portal/commit/11594965b60570fca9835aa8848e384d1eb942c5
- Pull https://github.com/liferay-frontend/liferay-portal/pull/570

`user_locale_options.jsp` was extracted from `portal-web` to its own module in LPS-115798 so it shouldn't be necessary to change it in master, I will remove that commit and force-push the changes.

**Comment by dantewang** [comment]
@joseluisbango Thanks for the update, but sorry I get a bit confused again by your reply about how the JSP is extracted.

I would like to get clarification about:

- is the change to this file needed for your fix on master?
- if not, can this change be considered as a part of "applying the new pattern"?

**Comment by joseluisbango** [comment]
Hi @dantewang. Let me clarify:

> - is the change to this file needed for your fix on master?

No, I don't think so. I think `portal-web/.../user_locale_options.jsp` is not used anymore after being extracted to `portal-user-locale-options-web`.

> - if not, can this change be considered as a part of "applying the new pattern"?

I would keep that file without any modification in master and then add an additional commit when backporting to 7.2.

Does it sound ok to you? Sorry for the confusion.

File: modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/servlet/FriendlyURLServlet.java [hunk 1]
```diff
@@ -284,7 +285,9 @@ public Redirect getRedirect(
 
 				if (Validator.isNotNull(i18nLanguageId) &&
 					!LanguageUtil.isAvailableLocale(
-						group.getGroupId(), i18nLanguageId)) {
+						group.getGroupId(), i18nLanguageId) &&
+					(!PortalUtil.isControlPanelPath(path) ||
```
Comment (dantewang) [inline modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/servlet/FriendlyURLServlet.java]: @joseluisbango Please use `portal.isControlPanelPath` here. The `portal` is already a `@Reference` in this class (all its sub-classes are OSGi components).

Comment (dantewang) [reply modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/servlet/FriendlyURLServlet.java]: Also in https://github.com/liferay-core-infra/liferay-portal/pull/961/commits/7bb6c9480ad81ffbbf19601dd57b3ee92764124a

**Review by dantewang — CHANGES_REQUESTED** [review]
@joseluisbango 
Please modify your commit according to the comments. By the way, I want to clarify some of the behavior:

I built on this PR and started DXP, changed the admin's language to Chinese, and created a site with only English as default and available language.

- Since I didn't enable the portal property you added, I reproduced the behavior (the site is displayed as English after creating a page); however,
- after shutting down portal, changing the property to `true`, and restarting, the behavior became:
  - English was displayed when I opened localhost:8080
  - I was able to switch back to Chinese by going to Account Settings and clicking on the notification
  - However, every time I went to any of the site pages, or opening the content page editor, they are displayed as English,
  - and when I went back to control panel page of the site, I got a notification which allows me to switch back to Chinese

What I want to clarify:
- Is the behavior expected that I have to manually change the language every time after I visit a site page)?
- Is the behavior expected that the content page editor is still displayed in English?

**Comment by dantewang** [comment]
@joseluisbango By the way, before resending or force pushing, please make sure you run `ant all` in your local, and commit bnd.bnd/packageinfo changes in `portal-kernel` and `portal-impl` related to what you modified if they exist. Thanks

**Comment by joseluisbango** [comment]
Hi @dantewang,

Sure! It sound reasonable to me too. I have already edited my previous commits and removed the unnecessary one. 

Thanks for the review!

**Review by dantewang — APPROVED** [review]

**Comment by dantewang** [comment]
ci:test:sf

**Comment by dantewang** [comment]
ci:test:relevant

**Comment by dantewang** [comment]
ci:test

**Comment by dantewang** [comment]
@tinatian This one looks OK to me.

**Comment by zhangyan0701** [comment]
No regressions found.

**Comment by tinatian** [comment]
@joseluisbango This is related to the behavior of user, I believe this is in the user&system team scope, could you please send pull to them to verify? I am going to close this pull, please feel free to reopen if user&system team does not take it.

Thanks.
Tina

**Comment by tinatian** [comment]
@joseluisbango Also, please add test to cover your changes.

Thanks.
Tina.

**Pull request closed without merging**